enhancements/srvkp-2021-downstream-osp-on-pr.md

```diff
@@ -0,0 +1,153 @@
+# … and the name of the pipelinerun to watch the execution
+# […]
+# Building in the background
+# […]
+# Once the PipelineRun is done (how we know, TBD)
+$ operator-sdk run -n openshift-operators bundle \
+    quay.io/openshift-pipeline/openshift-pipelines-operator-bundle:test-my-thing-72a32c
+```
+
+### Details
+
+#### How downstream *build* works
+
+The *source of truth* today is in [openshift-pipelines gerrit](https://code.engineering.redhat.com/gerrit/plugins/gitiles/openshift-pipelines/).
+
+- `upstream_source.yaml` ([1.7 example](https://code.engineering.redhat.com/gerrit/plugins/gitiles/openshift-pipelines/+/refs/heads/pipelines-1.7-rhel-8/upstream_sources.yaml)) is where we track code from upstream that has to be built.
+  *This is one part of what we want to change*
+- `distgit/containers` is where the `Dockerfile` of all images needed are defined.
+- `Makefile` has the target to build a downstream operator based on `upstream_sources.yaml` (and the content of those sources)
+
+The usual flow is
+
+```bash
+# Make sure upstream is up-to-date
+$ make sources
+# Fetch operator payload
+$ make sources/operator/fetch-payload
+# Build and push the images as well as the bundle
+# … grab a coffe ☕, it can be long…
+$ make REMOTE=quay.io/vdemeest bundle/push
+# Run the operator on your OCP cluster
+$ make REMOTE=quay.io/vdemeest bundle/run
+```
+
+#### `Pipeline` work
+
+In a nutshell, the pipeline need to do the following
+
+- Update `upstream_source.yaml` based on user inputs
+- Generate payload from user inputs
+  *This part is tricky*
+    - Generate upstream payload
+      e.g. for `tektoncd/pipeline#pull/2345` we would run `ko reslove -f config` on the source to generate the payload (even if we don't use the generated image laters)
+    - Use the generated payload in the operator *fetch-payload* (`sources/operator/fetch-payload` target)
+      The PR [tektoncd/operator#641](https://github.com/tektoncd/operator/pull/641) should help there.
+- Build images and the bundle images
+
+Some of those can be parallelized, some might not. It also depends if we want "one task to build all `make REMOTE=… bundle/push`" (like our current nightly) or if we want to split the load into several tasks (*might require the matrix feature / custom task*).
+
+
+#### Service
+
+There is two options available here:
+
+1. Using `tektoncd/triggers`, similar to pipelines-as-code
+   *And maybe an interceptor (?)*
+   - Need an interceptor to transform the payload
+   - *or* need a Task that translate it into results to be used in the next tasks
+2. Our own service
+   - Written in Go
+   - Served using knative/serving
+
+Most likely, (1) is the most "tekton" native option (and we can still use knative serving).
+
+
+## Open Questions
+
+- What about *patches* ? What if in downstream we have a patch that is render *nil* by the PR we are trying to build ? Should we have an option to *bypass* the patches (all, one, …?).
```

Comment:
this will be nice to have, i wouldn't say it is a necessity in the first iteration. 🧑‍💻